sdk/java/pulumi/src/main/java/com/pulumi/resources/Resource.java

```diff
@@ -311,6 +300,25 @@ private static ImmutableMap<String, ProviderResource> convertToProvidersMap(@Nul
         return result.build();
     }
 
+    private static List<Output<String>> computeAliases(String name, String type, ResourceOptions options) {
```

Comment:
This "looks" right but I'm not super sure; it would be absolutely lovely if we could port unit tests for the alias computation in addition to the behavior integration tests. I found an example here:

https://github.com/pulumi/pulumi/blob/master/sdk/python/lib/test/langhost/aliases/__main__.py#L17

That would also be a great hint to maintainers on what this function is supposed to do.

Reply:
Makes sense 👍.

Is this a must have now or can I create an issue for this and do the work in a separate PR?

Reply:
Separate PR should be fine 🙇